middleware/src/main/java/com/nytimes/android/external/store3/middleware/GsonParserFactory.kt

```diff
@@ -0,0 +1,65 @@
+package com.nytimes.android.external.store3.middleware
+
+
+import com.google.gson.Gson
+import com.nytimes.android.external.store3.base.Parser
+
+import java.io.Reader
+import java.lang.reflect.Type
+
+import okio.BufferedSource
+
+/**
+ * Factory which returns various Gson [Parser] implementations.
+ */
+object GsonParserFactory {
+
+    /**
+     * Returns a new Parser which parses from [Reader] to the specified type, using
+     * a new default configured [Gson] instance.
+     */
+    fun <T> createReaderParser(type: Type): Parser<Reader, T> {
+        return createReaderParser(Gson(), type)
```

Comment:
All of the single expression bodies in the methods of this file can be converted to assignments :)

Reply:
done!